### ISSUE TITLE
stderr vs stdout

### ISSUE THREAD

**Issue by lucasccordeiro** [esbmc/esbmc#2082]
An ESBMC user asked whether we can change how we print our messages to the output. In particular, this user claims we should output our messages to the stdout instead of stderr.

**Comment by rafaelsamenezes** [comment]
This could be implemented but this design was based on how compilers (gcc, clang, intel) also throw their messages and warnings in stderr. The idea is that the output of esbmc in stdout regards verification results while stderr is for warnings and messages.

**Comment by sshivaji** [comment]
I will explain the problem in more detail.

If I run `esbmc --help |grep goto`, it does not work, it returns the whole result as it goes to standard error.

If I change the command to `esbmc --help 2>&1|grep goto`, it works. This means that all the help output is going to standard error and not standard out.

If I use gcc --help, it works correctly, e.g.
`gcc --help|grep subsequent` returns the right output below:
  -x <language>           Treat subsequent input files as having type <language>

**Comment by rafaelsamenezes** [comment]
Yeah, --help should probably go through stdout by default.

**Comment by sshivaji** [comment]
I definitely agree! I have more examples of oddness, for example GOTO program creation log message and VERIFICATION SUCCESSFUL goes to stderr. However, that is the next thing to worry about, help is useful to start with because there are so many options and my memory is not that good :)

**Comment by mikhailramalho** [comment]
Maybe we should make only log_error go to stderr, maybe log_warning?

**Comment by sshivaji** [comment]
Yes, log_error and log_warning can go to standard error, this is common practice. The rest should probably go to standard out, for example --help, Generated GOTO Program, VERIFICATION SUCCESSFUL etc. It will make it easier to use.

**Comment by lucasccordeiro** [comment]
@Yiannis128: could you please implement the suggestions for this issue?

**Comment by Yiannis128** [comment]
Hi, will be looking into implementing this. Do you want this behaviour to be controlled via some flag?

**Comment by lucasccordeiro** [comment]
Not necessarily.

**Comment by sshivaji** [comment]
Sorry, my time zone is in California, USA. We don't need flags for this, just the default behavior can be improved.

In fact, once you start the task, let me know. I have other places we can fix the code as well.

**Comment by Yiannis128** [comment]
> Sorry, my time zone is in California, USA. We don't need flags for this, just the default behavior can be improved.
> 
> In fact, once you start the task, let me know. I have other places we can fix the code as well.

Please let me know what places!

**Comment by Yiannis128** [comment]
> Sorry, my time zone is in California, USA. We don't need flags for this, just the default behavior can be improved.
> 
> In fact, once you start the task, let me know. I have other places we can fix the code as well.

Hi, I have created a PR for this and are implementing it. My apologies for taking so long, it happens sometimes when there's too many tasks. I would like to invite you to participate in our Zulip chat and have a conversation with us more personally if you would like :)

PR: https://github.com/esbmc/esbmc/pull/4153

Link: https://systemsandsoftwaresecurity.zulipchat.com/join/6ajabxaurkcwl6tl3ify55jv/
